File: dotlox/Token.cs
namespace dotlox
{
    public enum TokenType
    {
        // Single Char tokens
        LEFT_PAREN,
        RIGHT_PAREN,
        LEFT_BRACE,
        RIGHT_BRACE,
        COMMA,
        DOT,
        MINUS,
        PLUS,
        SEMICOLON,
        SLASH,
        STAR,

        // One OR Two Char tokens
        BANG,
        BANG_EQUAL,
        EQUAL,
        EQUAL_EQUAL,
        GREATER,
        GREATER_EQUAL,
        LESS,
        LESS_EQUAL,

        // Literals
        IDENT,
        STRING,
        NUMBER,

        // Keywords
        FALSE,
        TRUE,
        AND,
        OR,
        CLASS,
        FUN,
        IF,
        ELSE,
        FOR,
        WHILE,
        NIL,
        RETURN,
        PRINT,
        SUPER,
        THIS,
        VAR,

        EOF
    }

    public class Token
    {
        TokenType type;
        String lexeme;
        Object literal;
        int line;

        public Token(TokenType type, String lexeme, Object literal, int line)
        {
            this.type = type;
            this.lexeme = lexeme;
            this.literal = literal;
            this.line = line;
        }

        public override String ToString()
        {
            return $"{type} {lexeme} {literal}";
        }

    }


    class Keywords: Dictionary<String, TokenType> {}

}

File: plox/plox/Token.py
from enum import StrEnum, auto
from dataclasses import dataclass


class TokenType(StrEnum):
    # Single Char tokens
    LEFT_PAREN = auto()
    RIGHT_PAREN = auto()
    LEFT_BRACE = auto()
    RIGHT_BRACE = auto()
    COMMA = auto()
    DOT = auto()
    MINUS = auto()
    PLUS = auto()
    SEMICOLON = auto()
    SLASH = auto()
    STAR = auto()

    # One OR Two Char tokens
    BANG = auto()
    BANG_EQUAL = auto()
    EQUAL = auto()
    EQUAL_EQUAL = auto()
    GREATER = auto()
    GREATER_EQUAL = auto()
    LESS = auto()
    LESS_EQUAL = auto()

    # Literals
    IDENT = auto()
    STRING = auto()
    NUMBER = auto()

    # Keywords
    FALSE = auto()
    TRUE = auto()
    AND = auto()
    OR = auto()
    CLASS = auto()
    FUN = auto()
    IF = auto()
    ELSE = auto()
    FOR = auto()
    WHILE = auto()
    NIL = auto()
    RETURN = auto()
    PRINT = auto()
    SUPER = auto()
    THIS = auto()
    VAR = auto()
    EOF = auto()


@dataclass
class Token:
    type_: TokenType
    lexeme: str
    literal: object
    line: int


Keywords = {
    "add": TokenType.AND,
    "class": TokenType.CLASS,
    "else": TokenType.ELSE,
    "false": TokenType.FALSE,
    "for": TokenType.FOR,
    "fun": TokenType.FUN,
    "if": TokenType.IF,
    "nil": TokenType.NIL,
    "or": TokenType.OR,
    "print": TokenType.PRINT,
    "return": TokenType.RETURN,
    "super": TokenType.SUPER,
    "this": TokenType.THIS,
    "true": TokenType.TRUE,
    "var": TokenType.VAR,
    "while": TokenType.WHILE,
}
